### ISSUE TITLE
Javascript "for-of" loop within arrow function IIFE => causing corruption in code formatting

### ISSUE THREAD

**Issue by dough654** [Microsoft/vscode#39837]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:32:36.023Z)
- OS Version: Windows_NT x64 6.1.7601
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
html-snippets|abu|0.1.0
cjsx|cri|0.0.6
vscode-eslint|dba|1.4.3
githistory|don|0.2.3
vscode-babel-coloring|dza|0.0.4
vue-snippets|hol|0.1.5
vue|liu|0.1.5
mssql|ms-|1.2.1
csharp|ms-|1.13.1
jsx|Twe|0.0.1

(1 theme extensions excluded)

---

When editing a javascript file, I have a for-of loop within an IIFE (as an arrow function). The editor does not interpret the formatting correctly. It produces red squiggles all over, and the auto indent command (alt + shift + F) is all over the place:

![image](https://user-images.githubusercontent.com/20626718/33689686-68ae7042-daae-11e7-891c-4882b4e030e2.png)

This code appears to be valid Javascript and runs as expected.

Without the IIFE is fine:

![image](https://user-images.githubusercontent.com/20626718/33689726-83c1ad0e-daae-11e7-9fc7-74e6261f87b5.png)

Changing the IIFE to not use an arrow function also works:

![image](https://user-images.githubusercontent.com/20626718/33689764-a3256cc6-daae-11e7-816d-7a7109e74963.png)

My jsconfig.json contents are:

```js
{
	// See go.microsoft.com/fwlink/?LinkId=759670
	// for the documentation about the jsconfig.json format
	"compilerOptions": {
		"target": "es2017",
		"module": "commonjs",
		"allowSyntheticDefaultImports": true
	},
	"exclude": [
		"node_modules",
		"bower_components",
		"jspm_packages",
		"tmp",
		"temp"
	]
}
```

Steps to Reproduce:

1. Create a new *.js file 
2. Paste the following code:

```js
(_ => {
    let arr = [
    'abc',
    'def'
]
    for (let x of arr) {
    console.log(x);
}   
})(); 
```

3. Run the auto-format command (alt + shift + F)

Reproduces without extensions: Yes

**Comment by mjbvz** [comment]
Thanks. This appears to have been fixed in typescript@next (TypeScript 2.7). 

We won't be picking up TypeScript 2.7 until January, but you can followed [these instructions](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions) to start using it today

**Comment by dough654** [comment]
Well that was quick. Thanks for the response
